transformer_engine/jax/cpp_extensions.py

```diff
@@ -1503,7 +1503,7 @@ def get_batch_per_block(k_seqlen: int) -> int:
         batches_per_warp = 2 if pow2 <= 128 else 1
         warps_per_block = threads_per_block / warp_size
         batches_per_block = warps_per_block * batches_per_warp
-        return batches_per_block
+        return int(batches_per_block)
```

Comment:
We can assume that `threads_per_block` is a multiple of `warp_size`:
```suggestion
        warps_per_block = threads_per_block // warp_size
        batches_per_block = warps_per_block * batches_per_warp
        return batches_per_block
```
There'll be other problems if the blocks are not made up of full warps. If we want to be absolutely certain, we could add something like `assert threads_per_block % warp_size == 0`, but I don't think it's necessary.